tests/simulate/test_dilute.py

```diff
@@ -0,0 +1,152 @@
+        self.assertDictEqual(d.options, {})
+
+        #no operations, with options
+        d = relentless.simulate.Dilute(**options)
+        self.assertCountEqual(d.operations, [])
+        self.assertDictEqual(d.options, options)
+
+        #with operations, with options
+        d = relentless.simulate.Dilute(operations, **options)
+        self.assertCountEqual(d.operations, operations)
+        self.assertDictEqual(d.options, options)
+
+    def test_run(self):
+        """Test run method."""
+        analyzer = relentless.simulate.dilute.AddEnsembleAnalyzer()
+        operations = [self.UpdateEnsemble(),self.UpdatePotential()]
+        ens = relentless.Ensemble(T=1.0, V=relentless.Cube(L=2.0), N={'A':2,'B':3})
+        pot = relentless.PairMatrix(types=ens.types)
+        for pair in pot:
+            pot[pair]['r'] = np.array([1.,2.,3.])
+            pot[pair]['u'] = np.array([2.,4.,6.])
+
+        #invalid ensemble (non-NVT)
+        ens_ = relentless.Ensemble(T=1, V=relentless.Cube(1), N={'A':2}, mu={'B':0.2})
+        d = relentless.simulate.Dilute(analyzer)
+        with self.assertRaises(ValueError):
+            d.run(ensemble=ens_, potentials=pot, directory=self.directory)
+
+        #invalid potential (r and u not defined)
+        pot_ = relentless.PairMatrix(types=ens.types)
+        with self.assertRaises(ValueError):
+            d.run(ensemble=ens, potentials=pot_, directory=self.directory)
+
+        #different run configurations
+        #no operations (other than analyzer)
+        d = relentless.simulate.Dilute(analyzer)
+        sim = d.run(ensemble=ens, potentials=pot, directory=self.directory)
+        ens_ = analyzer.extract_ensemble(sim)
+        self.assertAlmostEqual(ens_.T, 1.0)
+        self.assertAlmostEqual(ens_.V.volume, 8.0)
+        self.assertAlmostEqual(ens_.P, 0.2197740)
+        self.assertDictEqual(ens_.N.todict(), {'A':2,'B':3})
+        for pair in ens_.rdf:
+            np.testing.assert_allclose(ens_.rdf[pair].table,
+                                       np.array([[1.,np.exp(-2./1.)],
+                                                 [2.,np.exp(-4./1.)],
+                                                 [3.,np.exp(-6./1.)]]))
+
+        #with operations, no options
+        d = relentless.simulate.Dilute(operations + [analyzer])
+        sim = d.run(ensemble=ens, potentials=pot, directory=self.directory)
+        ens_ = analyzer.extract_ensemble(sim)
+        self.assertAlmostEqual(ens_.T, 3.0)
+        self.assertAlmostEqual(ens_.V.volume, 64.0)
+        self.assertAlmostEqual(ens_.P, 0.0302839)
+        self.assertDictEqual(ens_.N.todict(), {'A':3,'B':4})
+        for pair in ens_.rdf:
+            np.testing.assert_allclose(ens_.rdf[pair].table,
+                                       np.array([[2.,np.exp(-6./3.)],
+                                                 [4.,np.exp(-12./3.)],
+                                                 [6.,np.exp(-18./3.)]]))
```

Comment:
Tests like this can be removed if you move the ops above to `test_simulate.py`. Just check that the output is correct for one or two values of the ensemble.